### DISCUSSION TITLE
Location value cleared

### DISCUSSION THREAD

**Discussion by abolar** [netbox-community/netbox#9114] [General]
When i use the api for updating an existing device, using PUT, if the location is not in the payload, then netbox blanks out the location field. This behavior is not seen on any other fields -- meaning if the payload does not contain other fields (such as manufacturer or Region), netbox will not change the values for those fields. However, the region field gets blanked out.

**Comment by jeremystretch** [comment]
It might help to post your exact API request to tell what might be going wrong.
